### PR TITLE
Feature/interface tests

### PR DESCRIPTION
**Pull request by AlbertPG**

### CODE REVIEW

File: app/src/androidTest/java/org/pesmypetcare/mypetcare/activities/TestLoginActivity.java [hunk 1]
```diff
@@ -0,0 +1,101 @@
...
+@LargeTest
+public class TestLoginActivity {
+    public static final String LOG_IN = "LOG IN";
+    @Rule
+    public ActivityTestRule<LoginActivity> activityRule = new ActivityTestRule<>(LoginActivity.class);
+
+    @Test
+    public void shouldMakeTheSignUp() {
+        onView(withId(R.id.signUpUsernameText)).perform(typeText("johnDoe"), closeSoftKeyboard());
+        onView(withId(R.id.signUpMailText)).perform(typeText("johndoe@gmail.com"), closeSoftKeyboard());
```
Comment (santidrj) [inline app/src/androidTest/java/org/pesmypetcare/mypetcare/activities/TestLoginActivity.java]: Codacy Issue found: [The String "johndoe@gmail.com" appears 4 times in the file.](https://app.codacy.com/gh/Grupo13-PES-Mascotas/PES_My-Pet-Care/pullRequest?prid=5239997)

**Comment by santidrj** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
- Added 4
           

Complexity increasing per file
==============================
- app/src/androidTest/java/org/pesmypetcare/mypetcare/activities/fragments/TestRegisterPetFragment.java  1
- app/src/androidTest/java/org/pesmypetcare/mypetcare/activities/TestLoginActivity.java  1
- app/src/androidTest/java/org/pesmypetcare/mypetcare/activities/fragments/TestSettingsMenuFragment.java  1
- app/src/androidTest/java/org/pesmypetcare/mypetcare/activities/TestMainActivity.java  1
- app/src/androidTest/java/org/pesmypetcare/mypetcare/activities/fragments/TestNewPasswordFragment.java  1
         

Clones added
============
- app/src/androidTest/java/org/pesmypetcare/mypetcare/activities/fragments/TestNewPasswordFragment.java  2
         
```


See the complete overview on [Codacy](https://app.codacy.com/gh/Grupo13-PES-Mascotas/PES_My-Pet-Care/pullRequest?prid=5239997&bid=17176849)

File: app/src/androidTest/java/org/pesmypetcare/mypetcare/activities/TestLoginActivity.java [hunk 2]
```diff
@@ -0,0 +1,101 @@
...
+
+        onView(withId(R.id.signupButton)).perform(click());
+        onView(withId(R.id.signUpUsernameText)).check(matches(isDisplayed()));
+    }
+
+    @Test
+    public void shouldNotMakeTheSignUpIfThePasswordIsWeak() {
+        onView(withId(R.id.signUpUsernameText)).perform(typeText("johnDoe"), closeSoftKeyboard());
+        onView(withId(R.id.signUpMailText)).perform(typeText("johndoe@gmail.com"), closeSoftKeyboard());
+        onView(withId(R.id.signUpPasswordText)).perform(typeText("abc"), closeSoftKeyboard());
```
Comment (santidrj) [inline app/src/androidTest/java/org/pesmypetcare/mypetcare/activities/TestLoginActivity.java]: Codacy Issue found: [The String "abc" appears 3 times in the file.](https://app.codacy.com/gh/Grupo13-PES-Mascotas/PES_My-Pet-Care/pullRequest?prid=5239997)

File: app/src/androidTest/java/org/pesmypetcare/mypetcare/activities/TestLoginActivity.java [hunk 3]
```diff
@@ -0,0 +1,101 @@
...
+public class TestLoginActivity {
+    public static final String LOG_IN = "LOG IN";
+    @Rule
+    public ActivityTestRule<LoginActivity> activityRule = new ActivityTestRule<>(LoginActivity.class);
+
+    @Test
+    public void shouldMakeTheSignUp() {
+        onView(withId(R.id.signUpUsernameText)).perform(typeText("johnDoe"), closeSoftKeyboard());
+        onView(withId(R.id.signUpMailText)).perform(typeText("johndoe@gmail.com"), closeSoftKeyboard());
+        onView(withId(R.id.signUpPasswordText)).perform(typeText("AbcDert456$"), closeSoftKeyboard());
```
Comment (santidrj) [inline app/src/androidTest/java/org/pesmypetcare/mypetcare/activities/TestLoginActivity.java]: Codacy Issue found: [The String "AbcDert456$" appears 3 times in the file.](https://app.codacy.com/gh/Grupo13-PES-Mascotas/PES_My-Pet-Care/pullRequest?prid=5239997)

File: app/src/androidTest/java/org/pesmypetcare/mypetcare/activities/TestLoginActivity.java [hunk 4]
```diff
@@ -0,0 +1,101 @@
...
+@RunWith(AndroidJUnit4.class)
+@LargeTest
+public class TestLoginActivity {
+    public static final String LOG_IN = "LOG IN";
+    @Rule
+    public ActivityTestRule<LoginActivity> activityRule = new ActivityTestRule<>(LoginActivity.class);
+
+    @Test
+    public void shouldMakeTheSignUp() {
+        onView(withId(R.id.signUpUsernameText)).perform(typeText("johnDoe"), closeSoftKeyboard());
```
Comment (santidrj) [inline app/src/androidTest/java/org/pesmypetcare/mypetcare/activities/TestLoginActivity.java]: Codacy Issue found: [The String "johnDoe" appears 6 times in the file.](https://app.codacy.com/gh/Grupo13-PES-Mascotas/PES_My-Pet-Care/pullRequest?prid=5239997)

**Comment by santidrj** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
- Added 6
           

Complexity increasing per file
==============================
- app/src/androidTest/java/org/pesmypetcare/mypetcare/activities/fragments/TestSettingsMenuFragment.java  1
- app/src/androidTest/java/org/pesmypetcare/mypetcare/activities/fragments/TestNewPasswordFragment.java  1
- app/src/androidTest/java/org/pesmypetcare/mypetcare/activities/TestLoginActivity.java  1
- app/src/androidTest/java/org/pesmypetcare/mypetcare/activities/TestMainActivity.java  1
- app/src/androidTest/java/org/pesmypetcare/mypetcare/activities/fragments/TestRegisterPetFragment.java  1
         

Clones added
============
- app/src/androidTest/java/org/pesmypetcare/mypetcare/activities/fragments/TestNewPasswordFragment.java  2
         
```


See the complete overview on [Codacy](https://app.codacy.com/gh/Grupo13-PES-Mascotas/PES_My-Pet-Care/pullRequest?prid=5239997&bid=17176849)

File: app/src/androidTest/java/org/pesmypetcare/mypetcare/activities/fragments/TestRegisterPetFragment.java [hunk 5]
```diff
@@ -0,0 +1,97 @@
...
+    public static void setUpClass() {
+        MainActivity.setEnableLoginActivity(false);
+    }
+
+    @Test
+    public void shouldRegisterNewPet() {
+        onView(withId(R.id.flAddPet)).perform(click());
+        onView(withId(R.id.inputPetName)).check(matches(isDisplayed()));
+
+        onView(withId(R.id.inputPetName)).perform(typeText("Linux"), closeSoftKeyboard());
```
Comment (santidrj) [inline app/src/androidTest/java/org/pesmypetcare/mypetcare/activities/fragments/TestRegisterPetFragment.java]: Codacy Issue found: [The String "Linux" appears 2 times in the file.](https://app.codacy.com/gh/Grupo13-PES-Mascotas/PES_My-Pet-Care/pullRequest?prid=5239997)

File: app/src/androidTest/java/org/pesmypetcare/mypetcare/activities/TestLoginActivity.java (hunk 1 above)
Comment (santidrj) [inline app/src/androidTest/java/org/pesmypetcare/mypetcare/activities/TestLoginActivity.java]: Codacy Issue found: [The String "johndoe@gmail.com" appears 4 times in the file.](https://app.codacy.com/gh/Grupo13-PES-Mascotas/PES_My-Pet-Care/pullRequest?prid=5239997)

File: app/src/androidTest/java/org/pesmypetcare/mypetcare/activities/TestLoginActivity.java (hunk 4 above)
Comment (santidrj) [inline app/src/androidTest/java/org/pesmypetcare/mypetcare/activities/TestLoginActivity.java]: Codacy Issue found: [The String "johnDoe" appears 6 times in the file.](https://app.codacy.com/gh/Grupo13-PES-Mascotas/PES_My-Pet-Care/pullRequest?prid=5239997)

File: app/src/androidTest/java/org/pesmypetcare/mypetcare/activities/fragments/TestRegisterPetFragment.java [hunk 6]
```diff
@@ -0,0 +1,97 @@
...
+        onView(withId(R.id.inputGender)).perform(typeText("Male"), closeSoftKeyboard());
```
Comment (santidrj) [inline app/src/androidTest/java/org/pesmypetcare/mypetcare/activities/fragments/TestRegisterPetFragment.java]: Codacy Issue found: [The String "Male" appears 2 times in the file.](https://app.codacy.com/gh/Grupo13-PES-Mascotas/PES_My-Pet-Care/pullRequest?prid=5239997)

File: app/src/androidTest/java/org/pesmypetcare/mypetcare/activities/TestLoginActivity.java (hunk 3 above)
Comment (santidrj) [inline app/src/androidTest/java/org/pesmypetcare/mypetcare/activities/TestLoginActivity.java]: Codacy Issue found: [The String "AbcDert456$" appears 3 times in the file.](https://app.codacy.com/gh/Grupo13-PES-Mascotas/PES_My-Pet-Care/pullRequest?prid=5239997)

File: app/src/androidTest/java/org/pesmypetcare/mypetcare/activities/TestLoginActivity.java (hunk 2 above)
Comment (santidrj) [inline app/src/androidTest/java/org/pesmypetcare/mypetcare/activities/TestLoginActivity.java]: Codacy Issue found: [The String "abc" appears 3 times in the file.](https://app.codacy.com/gh/Grupo13-PES-Mascotas/PES_My-Pet-Care/pullRequest?prid=5239997)

**Review by enrichp — APPROVED** [review]

**Comment by santidrj** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
- Added 6
           

Complexity increasing per file
==============================
- app/src/androidTest/java/org/pesmypetcare/mypetcare/activities/fragments/TestRegisterPetFragment.java  1
- app/src/androidTest/java/org/pesmypetcare/mypetcare/activities/fragments/TestSettingsMenuFragment.java  1
- app/src/androidTest/java/org/pesmypetcare/mypetcare/activities/fragments/TestNewPasswordFragment.java  1
- app/src/androidTest/java/org/pesmypetcare/mypetcare/activities/TestLoginActivity.java  1
- app/src/androidTest/java/org/pesmypetcare/mypetcare/activities/TestMainActivity.java  1
         

Clones added
============
- app/src/androidTest/java/org/pesmypetcare/mypetcare/activities/fragments/TestNewPasswordFragment.java  2
         
```


See the complete overview on [Codacy](https://app.codacy.com/gh/Grupo13-PES-Mascotas/PES_My-Pet-Care/pullRequest?prid=5239997&bid=17176849)

File: app/src/androidTest/java/org/pesmypetcare/mypetcare/activities/TestLoginActivity.java (hunk 3 above)
Comment (santidrj) [inline app/src/androidTest/java/org/pesmypetcare/mypetcare/activities/TestLoginActivity.java:32]: Codacy Issue found: [The String "AbcDert456$" appears 3 times in the file.](https://app.codacy.com/gh/Grupo13-PES-Mascotas/PES_My-Pet-Care/pullRequest?prid=5239997)

File: app/src/androidTest/java/org/pesmypetcare/mypetcare/activities/fragments/TestRegisterPetFragment.java (hunk 5 above)
Comment (santidrj) [inline app/src/androidTest/java/org/pesmypetcare/mypetcare/activities/fragments/TestRegisterPetFragment.java:49]: Codacy Issue found: [The String "Linux" appears 2 times in the file.](https://app.codacy.com/gh/Grupo13-PES-Mascotas/PES_My-Pet-Care/pullRequest?prid=5239997)

File: app/src/androidTest/java/org/pesmypetcare/mypetcare/activities/TestLoginActivity.java (hunk 1 above)
Comment (santidrj) [inline app/src/androidTest/java/org/pesmypetcare/mypetcare/activities/TestLoginActivity.java:31]: Codacy Issue found: [The String "johndoe@gmail.com" appears 4 times in the file.](https://app.codacy.com/gh/Grupo13-PES-Mascotas/PES_My-Pet-Care/pullRequest?prid=5239997)

File: app/src/androidTest/java/org/pesmypetcare/mypetcare/activities/TestLoginActivity.java (hunk 4 above)
Comment (santidrj) [inline app/src/androidTest/java/org/pesmypetcare/mypetcare/activities/TestLoginActivity.java:30]: Codacy Issue found: [The String "johnDoe" appears 6 times in the file.](https://app.codacy.com/gh/Grupo13-PES-Mascotas/PES_My-Pet-Care/pullRequest?prid=5239997)

File: app/src/androidTest/java/org/pesmypetcare/mypetcare/activities/TestLoginActivity.java (hunk 2 above)
Comment (santidrj) [inline app/src/androidTest/java/org/pesmypetcare/mypetcare/activities/TestLoginActivity.java:52]: Codacy Issue found: [The String "abc" appears 3 times in the file.](https://app.codacy.com/gh/Grupo13-PES-Mascotas/PES_My-Pet-Care/pullRequest?prid=5239997)

File: app/src/androidTest/java/org/pesmypetcare/mypetcare/activities/fragments/TestRegisterPetFragment.java (hunk 6 above)
Comment (santidrj) [inline app/src/androidTest/java/org/pesmypetcare/mypetcare/activities/fragments/TestRegisterPetFragment.java:50]: Codacy Issue found: [The String "Male" appears 2 times in the file.](https://app.codacy.com/gh/Grupo13-PES-Mascotas/PES_My-Pet-Care/pullRequest?prid=5239997)

**Review by xavier-campos-diaz — APPROVED** [review]

**Pull request merged**